lib/server.js

```diff
@@ -262,23 +263,27 @@ function Server(opts, callback) {
   });
 
   that.on("subscribed", function(topic, client) {
-    that.publish({
-      topic: "$SYS/" + that.id + "/new/subscribes",
-      payload: JSON.stringify({
-        clientId: client.id,
-        topic: topic
-      })
-    });
+    if(that.modernOpts.publishSubscriptions) {
+      that.publish({
+        topic: "$SYS/" + that.id + "/new/subscribes",
+        payload: JSON.stringify({
+          clientId: client.id,
+          topic: topic
+        })
+      });
+    }
```

Comment:
you can avoid adding a listener on `subscribed`, instead of adding one and then doing nothing inside of it